src/ast.ml

```diff
@@ -15,99 +12,62 @@ let same_block_list_kind k1 k2 =
   | Bullet c1, Bullet c2 -> c1 = c2
   | _ -> false
 
-type link_def =
+type 'a def_elt =
   {
-    label: string;
-    destination: string;
-    title: string option;
-    attributes: attributes;
+    term: 'a;
+    defs: 'a list;
   }
 
-module type T = sig
-  type t
-end
-
-module MakeBlock (I : T) = struct
-  type def_elt =
-    {
-      term: I.t;
-      defs: I.t list;
-    }
-
-  and block =
-    {
-      bl_desc: block_desc;
-      bl_attributes: attributes;
-    }
-
-  and block_desc =
-    | Paragraph of I.t
-    | List of list_type * list_spacing * block list list
-    | Blockquote of block list
-    | Thematic_break
-    | Heading of int * I.t
-    | Code_block of string * string
-    | Html_block of string
-    | Definition_list of def_elt list
-end
-
-type link =
+type ('a, 'attr) block =
+  | Paragraph of 'attr * 'a
+  | List of 'attr * list_type * list_spacing * ('a, 'attr) block list list
+  | Blockquote of 'attr * ('a, 'attr) block list
+  | Thematic_break of 'attr
+  | Heading of 'attr * int * 'a
+  | Code_block of 'attr * string * string
```

Comment:
I think the only one that is confusing is the second string, perhaps we can alias a string type to a more descriptive name?

Reply:
Yep, that's right, just the two `string`s. It's not clear which is which (tho, imo, it's fairly natural to assume the first is the label and the second the code, and it'll be immediately obvious in nearly every case once there is actually content, so during pattern matching).

I think adding a type alias will cause confusing noise with type inference, since it'll lead other string values to show up as whatever the alias is named.

After sleeping on this a bit I'm  inclined to think we don't need to worry about this much: we can document it, and it's various straight forward. Happy to leave it up to @sonologico to do whatever he thinks best here.

Reply:
I agree that aliasing has some unfortunate consequences. I'll think about it a little more, but I'm also inclined to leave it as is.

Reply:
I've seen this in some other OCaml codebases. Didn't know about the unintended consequences. I think it's fine for now.